### PR TITLE
M4T2

### PR DESCRIPTION
**Pull request by levant47**

### CODE REVIEW

File: CartingService/API/Controller.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+    public Controller(ICartService cartService, IItemService itemService)
+    {
+        _cartService = cartService;
+        _itemService = itemService;
+    }
+
+    [ApiVersion(1, 2)]
+    [HttpGet("carts/{cartId}")]
+    public async Task<ActionResult> GetCart(int cartId)
```
Comment (rkosmirak) [inline CartingService/API/Controller.cs]: Shouldn't cart id be a string?

File: CartingService/API/Controller.cs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+    [ApiVersion(2)]
+    [HttpGet("carts/{cartId}/items")]
+    public async Task<ActionResult<List<ItemEntity>>> GetItems(int cartId)
+    {
+        var items = await _itemService.GetByCartId(cartId);
+        var routeValues = HttpContext.GetRouteData().Values.ToImmutableDictionary();
+        return Ok(new
+        {
+            _links = new { self = new { href = Url.Action() } },
```
Comment (rkosmirak) [inline CartingService/API/Controller.cs]: I'm not quite like these dynamic objects, as long as I have no idea what I can/cannot be added. Would you consider adding some generic view model (or something else) to make it consistent for different actions/controllers

File: CartingService/API/Program.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+{
+    options.SwaggerEndpoint("/swagger/v1/swagger.json", "v1");
+    options.SwaggerEndpoint("/swagger/v2/swagger.json", "v2");
+});
+
+app.Use(next => async context =>
+{
+    try { await next(context); }
+    catch (BadRequestException exception)
+    {
```
Comment (rkosmirak) [inline CartingService/API/Program.cs:44]: Not sure if it's ok. So, the client expects Json in regular cases, but Plain text if it's BadRequest?
Would you consider using the default bad request envelope from .net?

File: CartingService/BLL/ItemService.cs [hunk 4]
```diff
@@ -7,18 +7,30 @@
...
+        await _repository.Create(newEntity);
+        return newEntity;
     }
 
     // remove item from cart
-    public async Task Delete(int id)
+    public async Task<bool> Delete(int id)
     {
-        if (!await _repository.Exists(new() { Id = id })) { throw new BadRequestException($"Item with ID {id} does not exist"); }
+        if (!await _repository.Exists(new() { Id = id })) { return false; }
```
Comment (rkosmirak) [inline CartingService/BLL/ItemService.cs]: Personally, I'm not a big fan of the exceptions, but if you decided to use them, then do it consistently. Feel free to create a new exception (NotFoundException) and throw it here, so, it's consistent throughout the application.
Or consider implementing something like https://github.com/altmann/FluentResults
Some interesting insight can be found here https://www.youtube.com/watch?v=a1ye9eGTB98

File: CartingService/API/Controller.cs (hunk 2 above)
Comment (rkosmirak) [reply CartingService/API/Controller.cs]: Some interesting solution can be found here https://github.com/aloji/HATEOAS

File: CartingService/API/Controller.cs (hunk 1 above)
Comment (levant47) [reply CartingService/API/Controller.cs]: Fixed

File: CartingService/API/Program.cs (hunk 3 above)
Comment (levant47) [reply CartingService/API/Program.cs:44]: Fixed

File: CartingService/BLL/ItemService.cs (hunk 4 above)
Comment (levant47) [reply CartingService/BLL/ItemService.cs]: Fixed

File: CartingService/API/Controller.cs (hunk 2 above)
Comment (levant47) [reply CartingService/API/Controller.cs]: Fixed

**Review by rkosmirak — APPROVED** [review]

**Pull request merged**